src/util/Graph.java

```diff
@@ -14,22 +14,25 @@ public Graph(int rows, int columns) {
 		dimX = columns;
 		dimY = rows;
 	}
-	
+
 	public boolean haveIsolatedCores() {
-		ArrayList<Vertex> alc = new ArrayList<Vertex>();
-		//Escolha do 0.0 para ser o core inicial. Garantido a existencia do primeiro nodo em todas as topologias
-		vertex("0.0").checkIsolation(alc);
-		
-		//Se lista de alcancaveis for igual ao total de cores nao existe isolamento
-		if(!(alc.size()==vertices.size())) return true;
-		
-    	return false;
+		Set<Vertex> reachable = new HashSet<>();
+		checkIsolation(vertex("0.0"), reachable);
+		return !(reachable.size() == vertices.size());
+	}
+
+	private void checkIsolation(Vertex vertex, Set<Vertex> reachable) {
+		for (Edge adj : vertex.adjuncts()){
+			if (reachable.contains(adj.destination()))
```

Comment:
Put this condition outside the for (first thing on this method) checking for "vertex" and returning if it's already present.